### PR TITLE
networkd: re-assign prefix after DHCPv6 PD update

### PR DESCRIPTION
**Pull request by andir**
Previously on a lease update/renewal prefixes would not be announced by
radv anymore. Since PD prefixes can have a very long lifetime
(days/weeks/…) that issue wasn't really obvious in the previous (manual)
tests.

Instead of bailing out whenever there is already a PD prefix configured
on the interface we now let the code continue to re-assign the prefix as
long as it is the same prefix (as before).

This is a follow-up to #15669. During one of the iterations I added the code that checked if a prefix was already assigned to a link and then skipped those. This was the one difference that slipped in when compared to the version that I've been running for almost two months now.

### CODE REVIEW

File: src/network/networkd-dhcp6.c [hunk 1]
```diff
@@ -327,6 +324,14 @@ static int dhcp6_pd_prefix_distribute(Link *dhcp6_link,
                 } else if (r < 0)
                         return r;
 
+                /* Continue configuring the prefix if it is the first or the
+                 * same as already configured, otherwise just skip this link. */
+                if (dhcp6_link_get_dhcpv6_prefix(link, &current_prefix.in6) &&
+                                !in_addr_equal(AF_INET6, &current_prefix, &assigned_prefix)) {
```
Comment (yuwata) [inline src/network/networkd-dhcp6.c:330]: Please adjust the indentation.

File: src/network/networkd-dhcp6.c [hunk 2]
```diff
@@ -327,6 +324,14 @@ static int dhcp6_pd_prefix_distribute(Link *dhcp6_link,
...
+                                log_link_debug(link, "Already has another DHCPv6 prefix assigned.");
```
Comment (yuwata) [inline src/network/networkd-dhcp6.c:331]: `Another DHCPv6 prefix has been already assigned.`??

File: src/network/networkd-dhcp6.c [hunk 3]
```diff
@@ -1026,16 +1031,19 @@ static int dhcp6_prefix_remove_all(Manager *m, Link *link) {
         return 0;
 }
 
-static bool dhcp6_link_has_dhcpv6_prefix(Link *link) {
+static bool dhcp6_link_get_dhcpv6_prefix(Link *link, struct in6_addr *out) {
```
Comment (poettering) [inline src/network/networkd-dhcp6.c:1034]: we usually call return parameters "ret" or "ret_xyz"

File: src/network/networkd-dhcp6.c (hunk 2 above)
Comment (poettering) [inline src/network/networkd-dhcp6.c:331]: probably should say what is being done now. i.e. that this is ignored.

**Comment by keszybz** [comment]
@andir update?

**Comment by yuwata** [comment]
Now I am working on the issue in #16560.

**Comment by yuwata** [comment]
@andir I think the issue this PR tried to address is already fixed by #16585. Is it OK to close this PR?

**Comment by andir** [comment]
Yeah, I think so too.

**Comment by yuwata** [comment]
@andir Thanks. Let's close this.

**Pull request closed without merging**